src/ddmd/parse.d

```diff
@@ -5845,7 +5845,7 @@ final class Parser(AST) : Lexer
                 if (token.value == TOKfinally)
                 {
                     nextToken();
-                    finalbody = parseStatement(0);
+                    finalbody = parseStatement(PSscope);
```

Comment:
I had introduced a PSscope-related bug myself (https://github.com/dlang/dmd/pull/7023), so I thought I'd go hunting for more. :)

The fix for issue 17690 is here: https://github.com/dlang/dmd/pull/7026